dispatch/dispatch.go

```diff
@@ -161,20 +167,44 @@ func (d *Dispatcher) run(it provider.AlertIterator) {
 				return
 			}
 
-			level.Debug(d.logger).Log("msg", "Received alert", "alert", alert)
+			// this block is wrapped in a function to make sure that the span
+			// is ended before the next alert is processed
+			func() {
+				traceCtx, span := tracer.Start(d.ctx, "dispatch.Dispatcher.dispatch",
```

Comment:
(I thought I already responded to this but I guess my comment got lost somewhere 🤦‍♂️)

It might read nicer, but it'd imply that it's the span for the `run` method, which it isn't really as there may be many of these spans for a single call... maybe something like `handleAlert` would work?